src/pq_flash_index.cpp

```diff
@@ -563,29 +563,44 @@ LabelT PQFlashIndex<T, LabelT>::get_converted_label(const std::string &filter_la
 }
 
 template <typename T, typename LabelT>
-void PQFlashIndex<T, LabelT>::get_label_file_metadata(std::string map_file, uint32_t &num_pts,
+void PQFlashIndex<T, LabelT>::get_label_file_metadata(const std::string &file_content, uint32_t &num_pts,
                                                       uint32_t &num_total_labels)
 {
-    std::ifstream infile(map_file);
-    std::string line, token;
     num_pts = 0;
     num_total_labels = 0;
 
-    while (std::getline(infile, line))
+    size_t file_size = file_content.length();
+
+    std::string label_str;
+    size_t cur_pos = 0, next_pos = 0;
+
+    while (cur_pos < file_size && cur_pos != std::string::npos)
     {
-        std::istringstream iss(line);
-        while (getline(iss, token, ','))
+        next_pos = file_content.find('\n', cur_pos);
+        if (next_pos == std::string::npos)
```

Comment:
this looks like to drop the last line if the last line does not end with "\n"? I do not know if this is ok for current existing data. have you tested it?